src/Tomlyn/TomlModelOptions.cs

```diff
@@ -158,6 +188,42 @@ public TomlModelOptions()
         return ConvertPropertyName(name ?? prop.Name);
     }
 
+    /// <summary>
+    /// Default implementation for getting the field name
+    /// </summary>
+    private string? DefaultGetFieldNameImpl(FieldInfo field)
+    {
+        string? name = null;
+        foreach (var attribute in field.GetCustomAttributes())
+        {
+            var fullName = attribute.GetType().FullName;
+            // Check if attribute is ignored
+            foreach (var fullNameOfIgnoreAttribute in AttributeListForIgnore)
+            {
+                if (fullName == fullNameOfIgnoreAttribute)
+                {
+                    return null;
+                }
+            }
+
+            // Allow to dynamically bind to JsonPropertyNameAttribute even if we are not targeting netstandard2.0
```

Comment:
`JsonPropertyNameAttribute` supports `Field` as a target, so it makes sense to keep this.